staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -62,6 +62,42 @@ var (
 		},
 		[]string{"resource"},
 	)
+	watchersCounter = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Name: "apiserver_created_watchers",
+			Help: "Counter of watchers created in watchcache broken by resource type",
+		},
+		[]string{"resource"},
+	)
+	watchersGauge = metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Name: "apiserver_realtime_watchers",
+			Help: "Gauge of watchers watching in watchcache broken by resource type",
+		},
+		[]string{"resource"},
+	)
+	cacherEventsCounter = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Name: "apiserver_cacher_received_events",
+			Help: "Counter of events received from etcd broken by resource type",
+		},
+		[]string{"resource"},
+	)
+	eventsSentLatency = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Name:    "apiserver_cacher_events_sent_latency_milliseconds",
+			Help:    "Latency buckets of watchers sent in watchcache broken by resource type",
+			Buckets: metrics.DefBuckets,
+		},
+		[]string{"resource"},
+	)
+	watcherChannelLength = metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Name: "apiserver_cacher_watcher_channel_length",
+			Help: "Etcd watcher channel length",
+		},
```

Comment:
Do you mean API-server cache watcher channel length? Etcd watcher is the one you added below. We have a bunch of cache... watch... stuff in this area, It's easy to confuse people. To make it clearer, could you change this to  watcherChannelLength -> cacheWatcherChannelLength, and Name and Help accordingly?

Reply:
Refined:
```go
cacherWatcherChannelLength = metrics.NewGaugeVec(
		&metrics.GaugeOpts{
			Name: "apiserver_cacher_watcher_channel_length",
			Help: "Channel length of cacher watchers broken by resource type and channel name",
		},
		[]string{"resource", "channel"},
	)
```